src/get_programming_with_js_next/module_7_exercise.es6

```diff
@@ -0,0 +1,48 @@
+/*
+  Write a function that you can use to tag template literals that checks each
+  value if it is an object, and if so, converts the object into a string of
+  key/value pairs before interpolation.
+
+  Input:
+
+  const props = {
+    src: 'http://fillmurray.com/100/100',
+    alt: 'Bill Murray'
+  };
+
+  const img = withProps`<img ${props}>`;
+
+  Output:
+
+  <img src="http://fillmurray.com/100/100" alt="Bill Murray">
+*/
+
+const PROPS = {
+  src: 'http://fillmurray.com/100/100',
+  alt: 'Bill Murray'
+};
+
+function formatAttributes(attributes) {
+  let formattedAttributes = '';
+
+  for (let attribute in attributes) {
+    formattedAttributes += `${attribute}="${attributes[attribute]}" `;
+  }
+
+  return formattedAttributes.trim();
+}
+
+function withProps() {
+  const STRING_PARTS = arguments[0];
+  const PROPS = [].slice.call(arguments, 1).shift();
+
+  let htmlTag = STRING_PARTS.reduce(function (tag, part) {
```

Comment:
Prefer:
```javascript
const [openTag, closeTag] = strParts;
return `${openTag}${formatAttributes(props)}${closeTag}}`;
```

Rationale: Way more explicit - legibility
Though I do love your playing around with es6 features. And cleverness.
Additional note: I benchmarked string interpolation against string concatenation (using the `+` operator) and interpolation actually outperforms concatenation by a thin margin.

Reply:
Thanks for this suggestion. Learned that since `strParts` is coming in as an array, I can name its two parts with local variables. 👍 

You know I love `reduce`, but your suggestion is much simpler.